### PR TITLE
Fixes Shotguns being 100% broken

### LINKED ISSUE #9201: Bartenders shotgun cant reload
[Round ID]: # 31609
[Reproduction]: try and reload the double barrel shotgun when empty

### PR DESCRIPTION
**Pull request by Manly-Dwarf**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs )

:cl: Kawaii-Big-Boss
fix: You can now actually reload your shotgun.
fix: Contender still works too I guess
/:cl:

[why]: 
im sorry
im sorry carbonhell
please forgive me

Fixes #9201

### CODE REVIEW

**Comment by Manly-Dwarf** [comment]
Yes, this was actually thoroughly tested.
Originally all I really tested it on was the Revolver. I was stupid and thought that would be more than enough to prove there's no bugs and that the fix had no side effects. I ended putting out shitcode which got quick merged.

I tested with Revolvers first, as the Contender is technically a revolver, and they both work as usual. You can't reload more than you already have, and you can't swap in different ammo types.
I tested with Shotguns. You can reload them again. You can't shove in more than you max ammo, and you can only fit in shotgun shells.
I tested with Pistols. They were never broken, but I goofed off with magazines a bit and tried to see if I could break them. They worked just fine.
Contender also still works.

im sorry

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm [hunk 1]
```diff
@@ -50,16 +50,16 @@
...
-	// Hippie Edit - Makes the Contender and guns with universal ammo work by adding && caliber != all. Stop reverting this!
-	if ((!R || (caliber && R.caliber != caliber) || (!caliber && R.type != ammo_type)) && (caliber != "all")) /* hippie edit end */
+	// Hippie Start - Makes the Contender and guns with universal ammo work by adding && caliber != all. Stop reverting this!
+	if ((!R || (caliber && R.caliber != caliber) || (!caliber && R.type != ammo_type)) && (caliber != "all")) /* hippie end */
 		return FALSE
 
-/* hippie edit - Check for caliber == "all" */
-	if ((caliber == "all") && (stored_ammo.len < max_ammo))
+/* hippie start - Check for caliber == "all" */
+	if ((caliber == "all" && stored_ammo.len < max_ammo) || (stored_ammo.len < max_ammo))
```
Comment (Carbonhell) [inline code/modules/projectiles/boxes_magazines/_box_magazine.dm:58]: I don't get this. Is the caliber check actually needed here? I do not think so

**Comment by Carbonhell** [comment]
>a headmin merging a fix
Okay what the fuck are you guys doing now

**Comment by praisenarsie** [comment]
DAB ON CARBONHELL

**Comment by bubberducky2** [comment]
DAB ON CARBONHELL

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm (hunk 1 above)
Comment (Manly-Dwarf) [reply code/modules/projectiles/boxes_magazines/_box_magazine.dm:58]: Carbonhell, "caliber all" doesn't actually exist as a real caliber.
It's a fucking cheat in the if statement to force it in.

Comment (Carbonhell) [reply code/modules/projectiles/boxes_magazines/_box_magazine.dm:58]: the caliber == "all" condition has no effect, that is my point. Check yourself.

**Pull request merged**